test/helpers/testLib.js

```diff
@@ -7,39 +7,159 @@ const OldTellor = artifacts.require("./oldContracts/OldTellor.sol");
+
+const OldTellorStake = artifacts.require(
+  "tellorlegacy/contracts/oldContracts/libraries/OldTellorStake"
+);
+const OldTellorTransfer = artifacts.require(
+  "tellorlegacy/contracts/oldContracts/libraries/OldTellorTransfer"
+);
+const OldTellorDispute = artifacts.require(
+  "tellorlegacy/contracts/oldContracts/libraries/OldTellorDispute"
+);
+
+const OldTellorLibrary = artifacts.require(
+  "tellorlegacy/contracts/oldContracts/libraries/OldTellorLibrary"
+);
+
+const OldTellorGettersLibrary = artifacts.require(
+  "tellorlegacy/contracts/oldContracts/libraries/OldTellorGettersLibrary"
+);
+
+const V2TellorStake = artifacts.require("v2/libraries/v2TellorStake");
+const V2TellorTransfer = artifacts.require("v2/libraries/v2TellorTransfer");
+const V2TellorDispute = artifacts.require("v2/libraries/v2TellorDispute");
+const V2TellorLibrary = artifacts.require("v2/libraries/v2TellorLibrary");
+
+const prepareTellorTest = async () => {
+  const ttransfer = await TellorTransfer.new();
+
+  await TellorDispute.link(ttransfer);
+  const tdispute = await TellorDispute.new();
+
+  await TellorStake.link(tdispute);
+  await TellorStake.link(ttransfer);
+  const tstake = await TellorStake.new();
+
+  await TellorLibrary.link(ttransfer);
+  const tlib = await TellorLibrary.new();
+
+  await TellorLibraryTest.link(ttransfer);
+  const tlibtest = await TellorLibraryTest.new();
+
+  await Tellor.link(ttransfer);
+  await Tellor.link(tdispute);
+  await Tellor.link(tstake);
+  await Tellor.link(tlib);
+  await Tellor.link(tlibtest);
+};
+
+const prepareTellorV2 = async () => {
+  const v2ttransfer = await V2TellorTransfer.new();
+
+  await V2TellorDispute.link(v2ttransfer);
+  const v2tdispute = await V2TellorDispute.new();
+
+  await V2TellorStake.link(v2tdispute);
+  await V2TellorStake.link(v2ttransfer);
+  const v2tstake = await V2TellorStake.new();
+
+  await V2TellorLibrary.link(v2ttransfer);
+  const v2tlib = await V2TellorLibrary.new();
+
+  await TellorV2.link(v2ttransfer);
+  await TellorV2.link(v2tdispute);
+  await TellorV2.link(v2tstake);
+  await TellorV2.link(v2tlib);
+};
+
+const prepareOldTellor = async () => {
+  const oldttransfer = await OldTellorTransfer.new();
+
+  await OldTellorDispute.link(oldttransfer);
+  const oldtdispute = await OldTellorDispute.new();
+
+  await OldTellorStake.link(oldttransfer);
+  await OldTellorStake.link(oldtdispute);
+  const oldtstake = await OldTellorStake.new();
+
+  await OldTellorLibrary.link(oldttransfer);
+  const oldtlib = await OldTellorLibrary.new();
+
+  await OldTellor.link(oldttransfer);
+  await OldTellor.link(oldtdispute);
+  await OldTellor.link(oldtstake);
+  await OldTellor.link(oldtlib);
+
+  await TellorMaster.link(oldttransfer);
+  await TellorMaster.link(oldtstake);
+};
+
+async function prepareEnv() {
+  await prepareOldTellor();
+  await prepareTellorV2();
+  await prepareTellorTest();
+}
 
 async function mineBlock(env) {
```

Comment:
Mine block wasn't very deterministic, which made some variable test results. This changes make it more certain as well adds some information on the mined block, which is helpful for some test cases